connection.go

```diff
@@ -0,0 +1,115 @@
+package ts3
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	// DefaultPort is the default TeamSpeak 3 ServerQuery port.
+	DefaultPort = 10011
+
+	// DefaultSSHPort is the default TeamSpeak 3 ServerQuery SSH port.
+	DefaultSSHPort = 10022
+)
+
+// legacyConnection is an insecure TCP connection.
+type legacyConnection struct{ net.Conn }
+
+// Connect connects to the address with the given timeout.
+func (c *legacyConnection) Connect(addr string, timeout time.Duration) error {
+	addr, err := verifyAddr(addr, DefaultPort)
+	if err != nil {
+		return err
+	}
+
+	c.Conn, err = net.DialTimeout("tcp", addr, timeout)
+	return err
+}
+
+// sshConnection is an SSH connection with open SSH channel and attached shell.
+type sshConnection struct {
+	net.Conn
+	config  *ssh.ClientConfig
+	channel ssh.Channel
+}
+
+// Connect connects to the address with the given timeout and opens a new SSH channel with attached shell.
+func (c *sshConnection) Connect(addr string, timeout time.Duration) error {
+	addr, err := verifyAddr(addr, DefaultSSHPort)
+	if err != nil {
+		return err
+	}
+
+	if c.Conn, err = net.DialTimeout("tcp", addr, timeout); err != nil {
+		return err
+	}
+
+	clientConn, chans, reqs, err := ssh.NewClientConn(c.Conn, addr, c.config)
+	if err != nil {
+		return err
+	}
+	go ssh.DiscardRequests(reqs)
+
+	// reject all channel requests
+	go func(newChannel <-chan ssh.NewChannel) {
+		for channel := range newChannel {
+			_ = channel.Reject(ssh.Prohibited, ssh.Prohibited.String())
```

Comment:
Lets be explicit that we're ignoring errors.
```suggestion
			channel.Reject(ssh.Prohibited, ssh.Prohibited.String()) //nolint: errcheck
```